src/web/components/SignInGate/SignInGateSelector.tsx

```diff
@@ -144,6 +144,38 @@ const ShowSignInGate = ({
 	return <></>;
 };
 
+// custom hook can be used by other parts of DCR to determine if the sign in gate will or will not show
+export const useSignInGateWillShow = ({
+	isSignedIn,
+	CAPI,
```

Comment:
As a rule, we try to avoid passing `CAPI` as a prop because it is 1) enormous and 2) only vaguely typed. It's also easier to reason about what a thing does when it outlines the exact properties it depends on.

That said, it looks like you need to pass it in here because another function depends on it... Hmm, it's hard to know what to do here. It's not part of this PR but this PR highlights the problem